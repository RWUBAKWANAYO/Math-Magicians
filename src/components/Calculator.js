import React, { useState } from 'react';
import calculate from '../logic/calculate';
import Buttons from './Buttons';
import Display from './Display';

function Calculator() {
  const [state, setState] = useState({
    obj: {
      total: null,
      next: null,
      operation: null,
    },
  });

  const handleClick = (buttonName) => setState(({ obj }) => ({ obj: calculate(obj, buttonName) }));

  return (
    <div className="calculator-main-cont">
      <Display total={state.obj.total} next={state.obj.next} />
      <Buttons handleClick={handleClick} />
    </div>
  );
}

export default Calculator;
