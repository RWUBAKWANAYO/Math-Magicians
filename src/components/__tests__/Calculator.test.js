import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

it('Calculator page render', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
