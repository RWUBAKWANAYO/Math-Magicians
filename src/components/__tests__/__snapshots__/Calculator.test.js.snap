// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page render 1`] = `
<div
  className="calculator-cont"
>
  <h2>
    Let’s do some math !
  </h2>
  <div
    className="calculator-main-cont"
  >
    <div
      className="calculator-output"
    >
      0
    </div>
    <div
      className="calculator-wrapper"
    >
      <div
        className="calculator-numbers-cont"
      >
        <div
          className="calculator-numbers"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            %
          </button>
        </div>
        <div
          className="calculator-numbers"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            9
          </button>
        </div>
        <div
          className="calculator-numbers"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            6
          </button>
        </div>
        <div
          className="calculator-numbers"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            3
          </button>
        </div>
        <div
          className="calculator-numbers"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            .
          </button>
        </div>
      </div>
      <div
        className="calculator-symbol-cont"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          ×
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
